src/main/java/org/jnode/fs/ntfs/datarun/CompressedDataRun.java

```diff
@@ -162,105 +169,201 @@ public int readClusters(long vcn, byte[] dst, int dstOffset, int nrClusters, int
-            if (consumed == 0) {
-                // At the current point in time this is already zero but if the code
-                // changes in the future to reuse the temp buffer, this is a good idea.
-                uncompressedData.zero(0, uncompressed.length - uncompressedData.offset);
+            if (uncompressedChunkSize == 0) {
                 break;
             }
 
-            compressedData.offset += consumed;
-            uncompressedData.offset += BLOCK_SIZE;
+            compressedData.offset += compressedChunkSize;
+            uncompressedData.offset += uncompressedChunkSize;
         }
+
+        return uncompressedData.offset; //the actual length of the uncompressed data.
     }
 
     /**
-     * Uncompresses a single block.
+     * Decompresses a single block.
      *
      * @param compressed   the compressed buffer (in.)
      * @param uncompressed the uncompressed buffer (out.)
-     * @return the number of bytes consumed from the compressed buffer.
+     * @return the number of bytes consumed from the uncompressed buffer.
+     * @see <a href="https://learn.microsoft.com/en-us/openspecs/windows_protocols/ms-xca/cba0fa15-bd62-4eda-8838-8fc7ab406df1">LZNT1 Algorithm Details</a>
      */
-    private static int uncompressBlock(final OffsetByteArray compressed,
-                                       final OffsetByteArray uncompressed) {
+    private int decompressBlock(final OffsetByteArray compressed,
+                                final OffsetByteArray uncompressed) {
+
+        // nothing to read from the compressed array.
+        if (compressed.offset >= compressed.array.length) {
+            return 0;
+        }
+
+        // the current index in the uncompressed array. (based on the uncompressed.offset)
+        int pos = 0;
 
-        int pos = 0, cpos = 0;
+        // the current index in the compressed array. (based on the compressed.offset)
+        int cpos = 0;
 
+        // A compressed buffer consists of a series of one or more compressed output chunks. Each chunk begins with a 16-bit header.
+        // If both bytes of the header are 0, the header is an End_of_buffer terminal that denotes the end of the compressed data stream.
+        // Otherwise, the header MUST be formatted as follows:
+        //  Bit 15 indicates whether the chunk contains compressed data.
+        //  Bits [14:12] contain a signature indicating the format of the subsequent data.
+        //  Bits [11:0] contain the size of the compressed chunk, minus three bytes.
         final int rawLen = compressed.getShort(cpos);
         cpos += 2;
+
+        // Bits 14 down to 12 contain a signature value. This value MUST always be 3 (unless the header denotes the end of the compressed buffer).
+        final int signature = rawLen & 0x7000;
+        if (rawLen != 0 && signature != 0x3000 && log.isDebugEnabled()) {
+            log.error("ntfs_uncompblock: signature {} is not 3", signature);
+        }
```

Comment:
What is this code for.. should the log.isDebugEnabled() stay in the condition since you are using log.error?